profile.d/meson.build

```diff
@@ -4,15 +4,33 @@ install_data('70-systemd-shell-extra.sh', install_dir : shellprofiledir.startswi
 install_data('80-systemd-osc-context.sh', install_dir : shellprofiledir.startswith('/usr/') ? shellprofiledir : libexecdir / 'profile.d')
 
 if conf.get('LINK_SHELL_EXTRA_DROPIN') == 1
-        install_emptydir(shellprofiledir)
+        if meson.version().version_compare('>=1.3.0')
+                install_symlink(
+                        '70-systemd-shell-extra.sh',
+                        pointing_to : fs.relative_to(libexecdir / 'profile.d' / '70-systemd-shell-extra.sh',
+                                                     shellprofiledir),
+                        install_dir : shellprofiledir)
+        else
+                install_emptydir(shellprofiledir)
 
-        meson.add_install_script(sh, '-c',
-                ln_s.format(libexecdir / 'profile.d' / '70-systemd-shell-extra.sh', shellprofiledir / '70-systemd-shell-extra.sh'))
+                meson.add_install_script(sh, '-c',
+                        ln_s.format(libexecdir / 'profile.d' / '70-systemd-shell-extra.sh',
+                                    shellprofiledir / '70-systemd-shell-extra.sh'))
+        endif
 endif
```

Comment:
Hmm, does this make sense? Maybe let's just leave this as it was until we the baseline is moved to 1.3.0?

Reply:
If there's a perf advantage because of fewer shelling out it could be worth it, but should be measured

Reply:
I guess the more important advantage might be if we avoid noise in installation logs…

Reply:
Why not? I guess most developer uses newer meson, hence newer method can be used.
Newer method should be the way to go. But, we need to keep the old way for older meson.

Reply:
Yeah, this also minimizes adoption work in the future. I think we should actually make more use of the pattern.

Reply:
- using conditional code for newer versions of meson means that when dropping old versions in the future it is "obvious" what to do: simply drop the dead branch
- using `add_install_script` does not get tracked for `ninja uninstall`, whereas using `install_symlink` will get removed correctly